personality-insights/src/main/java/com/ibm/watson/developer_cloud/personality_insights/v3/PersonalityInsights.java

```diff
@@ -25,135 +22,115 @@
+    RequestBuilder builder = RequestBuilder.post("/v3/profile");
+    builder.query(VERSION, versionDate);
+    if (profileOptions.contentLanguage() != null) {
+      builder.header("Content-Language", profileOptions.contentLanguage());
     }
-
-    if (options.rawScores() != null) {
-      request.query(RAW_SCORES, options.rawScores());
+    if (profileOptions.acceptLanguage() != null) {
+      builder.header("Accept-Language", profileOptions.acceptLanguage());
     }
-
-    if (options.consumptionPreferences() != null) {
-      request.query(CONSUMPTION_PREFERENCES, options.consumptionPreferences());
+    builder.header("content-type", profileOptions.contentType());
+    if (profileOptions.rawScores() != null) {
+      builder.query("raw_scores", String.valueOf(profileOptions.rawScores()));
     }
-
-    if (options.language() != null) {
-      request.header(HttpHeaders.CONTENT_LANGUAGE, options.language());
+    if (profileOptions.csvHeaders() != null) {
+      builder.query("csv_headers", String.valueOf(profileOptions.csvHeaders()));
     }
-
-    if (options.acceptLanguage() != null) {
-      request.header(HttpHeaders.ACCEPT_LANGUAGE, options.acceptLanguage());
+    if (profileOptions.consumptionPreferences() != null) {
+      builder.query("consumption_preferences", String.valueOf(profileOptions.consumptionPreferences()));
     }
-
-    return request;
-  }
-
-  /**
-   * Accepts text and responds with a {@link Profile} with a tree of characteristics that include personality, needs,
-   * and values. <br>
-   * <br>
-   *
-   * @param text Text to analyze
-   *
-   * @return The personality {@link Profile}
-   */
-  public ServiceCall<Profile> getProfile(final String text) {
-    Validator.notEmpty(text, "text cannot be null or empty");
-
-    final ProfileOptions options = new ProfileOptions.Builder().text(text).build();
-    final RequestBuilder requestBuilder = buildProfileRequest(options);
-    return createServiceCall(requestBuilder.build(), ResponseConverterUtils.getObject(Profile.class));
-  }
-
-  /**
-   * Returns a {@link Profile} with a tree of characteristics that include personality, needs, and values. <br>
-   * <br>
-   * Here is an example of how to get the personality profile given some {@link ProfileOptions}:
-   *
-   * @param options the {@link ProfileOptions}
-   * @return The personality {@link Profile}
-   */
-  public ServiceCall<Profile> getProfile(final ProfileOptions options) {
-    final RequestBuilder requestBuilder = buildProfileRequest(options);
-    return createServiceCall(requestBuilder.build(), ResponseConverterUtils.getObject(Profile.class));
-  }
-
-  /**
-   * Returns a CSV profile.
-   *
-   * @param options the {@link ProfileOptions}
-   * @param includeHeaders if true returns the CSV headers
-   * @return the CSV profile
-   */
-  public ServiceCall<String> getProfileAsCSV(final ProfileOptions options, final boolean includeHeaders) {
```

Comment:
This is the kind of problem we need to fix either on the API design or the code generation.